cdap-docs/developers-manual/source/building-blocks/datasets/fileset.rst

```diff
@@ -130,3 +129,211 @@ See the Apache™ Twill®
 `API documentation <http://twill.incubator.apache.org/apidocs/org/apache/twill/filesystem/Location.html>`__
 for additional information about the ``Location`` abstraction.
 
+==================
+PartitionedFileSet
+==================
+
+While a FileSet is a convenient abstraction over actual file system interfaces, it still requires
+the application to be aware of file system paths. For example, an application that maintains data
+over time might have a new file for every month. One could come up with a naming convention that encodes
+the month into each file name, and share that convention across all applications that use this file set.
+Yet that can become tedious to manage, especially if the naming convention should ever change—then all
+applications would have to be changed simultaneously for proper functioning.
+
+The PartitionedFileSet Dataset relieves applications from understanding file name conventions. Instead,
+it associates a partition key with every file, for example the year and month associated with that file.
```

Comment:
Instead, it associates a partition key with every file; for example, the year and month associated with that file.